### DISCUSSION TITLE
Simultaneous Chats Limits / Like Queue

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#207] [Ideas & Features]
**Problems:**
- Overwhelming number of matches/messages for popular users, primarily women.
- Inefficient for all users: time wasted on unattainable matches, and genuine users lost in the crowd.

**Simultaneous Chat Limits:**
- **Limit the chats as people cannot possible manage more than a certain number of conversations at a single time**. This allows us to have a feature to help indicate that the user is currently preoccupied with talking to other people. This also helps users that want to talk to the person know they are busy and probably won't message you until they finish their queue (more predictable time).

**Liked Queue:**
- Limit the amount of people a profile can be liked *by* at a given time. This allows people to have a more manageable queue to go through when they log onto the app.
- Once the user hits the limit on the liked queue, their profile is no longer shown to other users. This helps to filter out people that are currently busy and popular profiles taking up all the attention on the app.

**Need to Solve:**

- Case where the user's matches are full, but someone from their queue adds them to their "Chat" list. (send user notification that expires after x days?)

**Potential Features:**
- Add "Liked Me" filter to search to view the people that currently like them to the people they can search for.

**Benefits:**
- Reduces wasted effort and user frustration by aligning efforts with potential rewards.
- Encourages meaningful connections by focusing on a manageable number of interactions.
- Addresses gender imbalance and improves the overall community experience.

**Comment by flo-kei** [comment]
Amazing feature, would love to see/implement this!

Something that should be resolved is the amount of people that can wait in the like-queue.
I'm not sure if 20 as suggested on Discord is a fitting number, I was more thinking of something lower such as 3-5.
A lower number would also ensure that people in that queue are not waiting forever to (maybe) be matched at some point. 

I also believe that this feature shouldn't add pressure/stress onto the user (e.g. I wouldn't tell them there are 20 people waiting in their like queue), because it could harm their interaction/chats with the users they are currently texting

**Reply by Oshyan** [reply]
I think 3-5 is an extremely small number. We need to find the balance between people not wasting time on people who are busy and the "busy" people still having enough Likes in their queue that they're likely to find a new match to replace anyone they decide to stop chatting/unmatch with. We could test different "depths" of Like Queue to see what works in the real world. And maybe it would even be good to consider a "dynamic" like queue depth/size, i.e. for people who are more active and go through their queue more frequently, they will have a larger queue. In other words maybe the depth of the queue could actually depend directly on how fast a given user goes through it, within limits, e.g. a queue depth between 5 and 20 or something. Interesting to think about, all subject to testing what actually works in the real world.

Letting people know when their queue is full is probably on balance a "positive" signal overall. We don't want people being fickle and jumping ship from current matches/conversations too quickly, but we also don't want users missing out on potentially good matches. It's another balancing act. Ultimately we have to trust the user to make the right decision for themselves.

**Comment by WilliamDormer** [comment]
I tend to like this idea.
